app/react/Library/components/FiltersForm.js

```diff
@@ -62,9 +62,14 @@ export class FiltersForm extends Component {
       documentTypes.get(0) || (templates.get(0) || fromJS({})).get('_id') || 'System';
     const allFields = this.props.fields.toJS();
     const showNoValueOnFilters = allFields.size;
-    const fields = libraryHelper
+    let fields = libraryHelper
```

Comment:
Given ```fields``` could be reassigned according the size of documentTypes, parseWithAggregations doesn't need to be called all the time. Couldn't it be a ternary assignment?

Reply:
The reassignment doesn't change what `parseWithAggregations` does, that helper needs to run every time the component renders since it does quite a bit. `prepareDefaultFilters` just adds on top when necessary.